src/actors/symbolication.rs

```diff
@@ -1075,6 +1083,75 @@ impl SymbolicationActor {
         Box::new(future)
     }
 
+    /// Join a procspawn handle with a timeout.
+    ///
+    /// This handles the procspawn result, makes sure to appropriately log any failures and
+    /// save the minidump for debugging.  Returns a simple result converted to the
+    /// `SymbolicationError`.
+    fn join_procspawn<T, E>(
+        handle: procspawn::JoinHandle<Result<procspawn::serde::Json<T>, E>>,
+        timeout: Duration,
+        metric: &str,
+        minidump: Bytes,
+        minidump_cache: crate::cache::Cache,
+    ) -> Result<T, SymbolicationError>
+    where
+        T: Serialize + DeserializeOwned,
+        E: Into<SymbolicationError> + Serialize + DeserializeOwned,
+    {
+        match handle.join_timeout(timeout) {
+            Ok(Ok(procspawn::serde::Json(out))) => Ok(out),
+            Ok(Err(err)) => Err(err.into()),
+            Err(perr) => {
+                let reason = if perr.is_timeout() {
+                    "timeout"
+                } else if perr.is_panic() {
+                    "panic"
+                } else if perr.is_remote_close() {
+                    "remote-close"
+                } else if perr.is_cancellation() {
+                    "canceled"
+                } else {
+                    "unknown"
+                };
+                let kind = if perr.is_timeout() {
+                    SymbolicationErrorKind::Timeout
+                } else {
+                    SymbolicationErrorKind::Canceled
+                };
+                metric!(counter(metric) += 1, "reason" => reason);
+                if let SymbolicationErrorKind::Canceled = kind {
+                    Self::save_minidump(minidump, minidump_cache)
+                        .map_err(|e| log::error!("Failed to save minidump {}", LogError(&e)))
+                        .map(|r| match r {
+                            Some(path) => sentry::configure_scope(|scope| {
+                                scope.set_tag("crashed_minidump", path.display());
+                            }),
```

Comment:
I ended up setting this as a tag, but I think I wanted a context?  Only context seems much more restrictive in the things which can be represented this way.  Is there good documentation somewhere that describes what these are for and their tradeoffs?

Reply:
Sorry, I missed that. `set_extra` is the way to go.